### ISSUE TITLE
improvements for Avalon build

### ISSUE THREAD

**Issue by adalke** [rdkit/rdkit#524]
I decided to install the Avalon component. I came across a few possible improvements to the installation instructions.

1) the installation documentation (eg, at https://github.com/rdkit/rdkit/blob/master/Docs/Book/Install.rst ) refers to "$RDBASE/External/AvalonTool/README". That should be the plural "AvalonTools" and not the singular form.

2) The README says "Extract the tar file somewhere". My suggestion is to have it download directly into the AvalonTools directory.

3) The README says to apply two diffs. It would be easier if those were in their own files instead of embedded in the README, and could be applied via "patch". Also, two of the edits concern blank lines and can be omitted. (For someone like me who ended up applying the patch manually.)

4) What about a "download-avalon.sh", to complement "download-inchi.sh", which does the download, untar, and patch in a single step?

5) Using cmake 3.2.2. I get the error message:

```
CMake Error at External/AvalonTools/CMakeLists.txt:2 (string):
  string sub-command REGEX, mode REPLACE needs at least 6 arguments total to
  commnd.
```

That line is:

```
string(REGEX REPLACE "\\\\" "/" AVALONTOOLS_DIR ${AVALONTOOLS_DIR})
```

This is because I didn't set AVALONTOOLS_DIR. However, if (2) is implemented then there can be a default value of $RDBASE/External/AvalonTool/ , which would mean one less configuration setting.

**Comment by bp-kelley** [comment]
I agree with Andrew that this is perhaps to complicated.  We could make the download part of the CMAKE scripts which may make an Avalon dependency on, say python 2.6 or higher, but that is probably not over the top.  I can make PR at some point in the future.

**Comment by greglandrum** [comment]
I completely agree

**Comment by rvianello** [comment]
I don't particularly like the download of auxiliary software components occurring at build time. It adds some complexity and reduces the flexibility of the cmake scripts, and it introduces an overhead in the build process, plus the possibility of builds failing because of network errors.

Maybe a cmake FindAvalon module could help in this case (although some issues should be probably addressed by the Avalon project itself), but I would actually prefer the addition of a separate download script as Andrew suggested, compared to the direct integration into the cmake build.

**Comment by thegodone** [comment]
Dear all,

It would be easier to include the avalon extension directly in the main
stream ?

BR,

Guillaume

2016-02-14 11:48 GMT+01:00 Riccardo Vianello <notifications@github.com>:

**Comment by bp-kelley** [comment]
@rvianello - I agree, however I'm unsure if we can/should have AVALON and/or INCHI as part of the repository, and at some point, someone has to download something to build, making it an optional download part of cmake doesn't seem like a terrible solution.  We already have to do this for the travis builds and it has been pretty robust.

**Comment by greglandrum** [comment]
Closing this because I believe it was resolved by #783.
Please reopen if that's not the case.
